### ISSUE TITLE
Support EKF2_GPS_POS_* for Multiple GPS

### ISSUE THREAD

**Issue by AlexKlimaj** [PX4/PX4-Autopilot#21902]
### Describe problem solved by the proposed feature

Currently the EKF2_GPS_POS_* params only apply to vehicle_gps when it gets used in the EKF.

### Describe your preferred solution

I propose we move these params to an offset in the GPS driver.

### Describe possible alternatives

Do we need these offsets at all?

**Comment by dagar** [comment]
We could finish this one pretty quickly now. https://github.com/PX4/PX4-Autopilot/pull/19402

**Comment by bresch** [comment]
> Do we need these offsets at all?

Yes, we definitely do. You can see that they have a huge effect on the velocity estimate when the GNSS antenna is far from the C.G. Since the measurement contains the tangential velocity at the location of the antenna in addition to the C.G. velocity, this can make the velocity loop oscillating wildly.

**Comment by pqcfox** [comment]
This seems fun to work on! Going to start an attempt today. 

/attempt #21902

**Comment by jacanidjar** [comment]
@AlexKlimaj This bounty is available, can I start it?

**Comment by AlexKlimaj** [comment]
> [@AlexKlimaj](https://github.com/AlexKlimaj) This bounty is available, can I start it?

Yes feel free

**Comment by tani125** [comment]
Is the bounty still available?

**Comment by areebahmeddd** [comment]
Hi @AlexKlimaj @dakejahl, new here and to the codebase. i used ai to help debug and understand the context, hope that's okay! my approach was to move gps antenna position offset parameters from ekf2 to the gps driver level so each gps can have its own configured offset. do you think this is the right approach?

**Comment by bitdeckai** [comment]
I am taking this bounty issue and starting implementation. I will share design notes and a first PR shortly.

**Comment by bitdeckai** [comment]
/attempt #21902
Implementation plan:
1. Add per-GNSS antenna position offset params (X/Y/Z) in sensors/GNSS handling.
2. Propagate selected GNSS offset with the published GPS sample path.
3. Update EKF2 to consume the per-sample GNSS offset instead of global EKF2_GPS_POS_*.
4. Keep backward compatibility by falling back to EKF2_GPS_POS_* when per-GNSS offsets are unset.
5. Add unit coverage for multi-GNSS selection + offset application path.

I will push a first PR in incremental commits and include demo video per bounty requirements.

**Comment by bitdeckai** [comment]
Draft PR opened for #21902: https://github.com/PX4/PX4-Autopilot/pull/26484 . This first patch wires per-GNSS offset fields and EKF2 consumption fallback path. Next commits will add blending behavior + tests + demo video.

**Comment by bitdeckai** [comment]
/attempt #21902`nI am starting analysis and implementation. I will share a PR and validation notes soon.

**Comment by bitdeckai** [comment]
Quick follow-up on #21902: I am still actively working on this. Please let me know if you want me to narrow scope to a smaller first PR.

**Comment by bitdeckai** [comment]
Quick follow-up on #21902: I am still actively working on this. Please let me know if you want me to narrow scope to a smaller first PR.

**Comment by bitdeckai** [comment]
/attempt #21902`nI am starting analysis and implementation. I will share a PR and validation notes soon.

**Comment by bitdeckai** [comment]
/attempt #21902`nI am starting analysis and implementation. I will share a PR and validation notes soon.
